app-cpp/app355.cpp

```diff
@@ -480,7 +369,388 @@ void SatOrbit::OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainV
         trainCount++;
     });
 }
+#else
+//----------------------------------------
+#pragma region SatOrbitSingle
+
+class SatOrbitSingle : public SatOrbit
+{
+public:
+    SatOrbitSingle() = default;
+    virtual ~SatOrbitSingle() = default;
+    static std::unique_ptr<SatOrbitSingle> Make();
+
+// Implementation
+private:
+    // SatOrbit
+    std::vector<sat355::TLE> OnReadFromFile(int argc, char* argv[]) override;
+    void OnCalculateOrbitalData(const std::vector<sat355::TLE>& inTLEVector, OrbitalDataVector& ioDataVector) override;
+    void OnSortOrbitalVector(std::vector<OrbitalData>& ioOrbitalVector) override;
+    std::vector<std::vector<OrbitalData>> OnCreateTrains(const std::vector<OrbitalData>& orbitalVector) override;
+    void OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainVector) override;
+
+// Helper
+protected:
+    static bool SortPredicate(const OrbitalData& inLHS, const OrbitalData& inRHS);
```

Comment:
Organize your classes by ordering their access control like this:
First: `public:`,
Then: `protected:`
Finally: `private:`

Note that your could have "more than 1" series of `public:` or `private:` to emphasize "groupings" of methods as part of some larger inherited interface.